frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -295,30 +321,23 @@ fn decl_runtime_metadata<'a>(
 	extrinsic: &TypePath,
 ) -> TokenStream2 {
 	let modules_tokens = module_declarations
-		.filter_map(|module_declaration| {
-			module_declaration.find_part("Module").map(|_| {
-				let filtered_names: Vec<_> = module_declaration
-					.module_parts()
-					.into_iter()
-					.filter(|part| part.name() != "Module")
-					.map(|part| part.ident())
-					.collect();
-				(module_declaration, filtered_names)
-			})
-		})
-		.map(|(module_declaration, filtered_names)| {
+		.map(|module_declaration| {
+			let parts = module_declaration.module_parts().into_iter()
+				.map(|part| part.ident());
+
 			let module = &module_declaration.module;
 			let name = &module_declaration.name;
-			let instance = module_declaration
-				.instance
-				.as_ref()
-				.map(|name| quote!(<#name>))
-				.into_iter();
+			let instance = module_declaration.instance.as_ref()
+				.map(|name| quote!(<#name>));
+
+			let pallet_or_module = &module_declaration.pallet_or_module;
 
 			let index = module_declaration.index;
 
 			quote!(
-				#module::Module #(#instance)* as #name { index #index } with #(#filtered_names)*,
+				// Note: we can keep the parts Pallet or Module or ValidateUnsigned because they
+				// are not used by impl_runtime_metadata anyway.
```

Comment:
What does this mean?

Reply:
below it calls impl_runtime_metadata, previously it was filtering out the "part" Module (by part I mean the stuff like Module, ValidateUnsigned, Call, Event, Config or `Config<T>`, ... given to each pallet).
But there is really no need to do so, impl_runtime_metadata can be given all the parts without any issue.

I'll reformulate the comment